gstd/src/common/handlers.rs

```diff
@@ -25,50 +25,253 @@
+            use crate::prelude::fmt::Write;
+            use arrayvec::ArrayString;
+
+            static_assertions::const_assert!(PANICKED_AT.len() == (PANIC_OCCURRED.len() - 4));
+
+            let mut debug_msg = ArrayString::<{ PANIC_OCCURRED.len() + TRIMMED_MAX_LEN }>::new();
+
+            let _ = debug_msg.try_push_str(&PANIC_OCCURRED[..4]);
+            let _ = write!(&mut debug_msg, "{panic_info}");
+
+            // SAFETY: `debug_msg.len() >= PANIC_OCCURRED.len()` because `try_push_str`
+            // pushes string `"pani"` and `write!()` pushes string `"panicked at "`.
+            // The capacity of `debug_msg` is always enough to do this.
+            unsafe {
+                debug_msg
+                    .as_bytes_mut()
+                    .get_unchecked_mut(4..PANIC_OCCURRED.len())
+                    .copy_from_slice(&PANIC_OCCURRED[4..].as_bytes());
+            }
+
+            let _ = ext::debug(&debug_msg);
+
+            // SAFETY: `debug_msg` is guaranteed to be initialized since `try_push_str` does
+            // `memcpy` (see panic handler for nightly rust for more details).
+            let msg = unsafe { debug_msg.get_unchecked(PANIC_OCCURRED.len()..) };
+            ext::panic(&msg)
+        }
+
+        /// Panic handler for stable Rust >=1.73.
+        #[rustversion::since(1.73)]
+        #[cfg(feature = "debug")]
+        #[cfg(not(feature = "panic-messages"))]
+        #[panic_handler]
+        pub fn panic(panic_info: &PanicInfo) -> ! {
+            use crate::prelude::{fmt::Write, str};
+            use arrayvec::ArrayString;
+
+            let mut default_panic_msg =
+                ArrayString::<{ PANICKED_AT.len() + TRIMMED_MAX_LEN }>::new();
+            let _ = write!(&mut default_panic_msg, "{panic_info}");
+
+            fn parse_panic_msg(msg: &str) -> Option<(&str, &str)> {
+                const NEEDLE: [u8; 2] = *b":\n";
+
+                // SAFETY: We can use `str::from_utf8_unchecked` because the delimiter is ascii
+                // characters. Therefore, the strings between the delimiter will be a valid
+                // UTF-8 sequence (see https://en.wikipedia.org/wiki/UTF-8#Encoding).
+                let msg_bytes = msg.as_bytes();
+                msg_bytes
+                    .windows(NEEDLE.len())
+                    .position(|window| NEEDLE.eq(window))
+                    .map(|pos| unsafe {
+                        (
+                            str::from_utf8_unchecked(
+                                msg_bytes.get_unchecked(PANICKED_AT.len()..pos),
+                            ),
+                            str::from_utf8_unchecked(
+                                msg_bytes.get_unchecked((pos + NEEDLE.len())..),
+                            ),
+                        )
+                    })
+            }
+
+            let option = parse_panic_msg(&default_panic_msg);
+            let (location, message) = unsafe { option.unwrap_unchecked() };
```

Comment:
It can abort without panic message if `location` is very long.

Reply:
Abort is not recommended. Call of `gr_panic` should occur at any circumstances

Reply:
f7e459bfe94c07abde45616da15d993728a6dd99